zcash_primitives/src/primitives.rs

```diff
@@ -121,10 +118,16 @@ impl Diversifier {
     }
 }
 
+/// A Sapling payment address.
+///
+/// # Invariants
+///
+/// `pk_d` is guaranteed to be prime-order (i.e. in the prime-order subgroup of Jubjub,
+/// and not the identity).
 #[derive(Clone, Debug)]
 pub struct PaymentAddress<E: JubjubEngine> {
-    pub pk_d: edwards::Point<E, PrimeOrder>,
-    pub diversifier: Diversifier,
+    pk_d: edwards::Point<E, PrimeOrder>,
```

Comment:
Here I will add my regularly scheduled exhortation to rename the `PrimeOrder` marker to `PrimeSubgroup` (https://github.com/zcash-hackworks/sapling-crypto/issues/97). Here we actually have a point that is enforced to be prime order, and renaming the marker is *still* the right thing to do, because it's not the marker that is enforcing that.

Does not block.